pulp_container/tests/functional/api/test_sync.py

```diff
@@ -1,231 +1,125 @@
-        client_api = gen_container_client()
-        cls.repositories_api = RepositoriesContainerApi(client_api)
-        cls.remotes_api = RemotesContainerApi(client_api)
-        cls.tags_api = ContentTagsApi(client_api)
-        cls.versions_api = RepositoriesContainerVersionsApi(client_api)
+    return _synced_repository_with_filtered_tags
 
-        cls.repository = None
 
-    def test_sync_with_non_existing_included_tag(self):
-        """Check whether the sync machinery ignores a non-existing tag."""
-        include_tags = ["manifest_a", "non_existing_manifest"]
-        self.sync_repository_with_filtered_tags(include_tags)
+@pytest.mark.parallel
+def test_basic_sync(
+    container_repo, container_remote, container_repository_api, container_tag_api, sync
+):
 
-        self.assert_synced_tags(["manifest_a"])
+    sync(container_repo, container_remote)
+    repository = container_repository_api.read(container_repo.pulp_href)
+    first_sync_tags = container_tag_api.list(
+        repository_version=repository.latest_version_href
+    ).results
 
-    def test_sync_with_included_tags(self):
-        """Test whether the repository is synced only with included tags."""
-        include_tags = ["manifest_a", "manifest_b", "manifest_c"]
-        self.sync_repository_with_filtered_tags(include_tags)
+    assert "versions/1/" in repository.latest_version_href
 
-        self.assert_synced_tags(include_tags)
+    latest_version_href = repository.latest_version_href
+    sync(
+        container_repo, container_remote
+    )  # We expect that this second sync doesn't create a new repo version
+    repository = container_repository_api.read(repository.pulp_href)
+    second_sync_tags = container_tag_api.list(
+        repository_version=repository.latest_version_href
+    ).results
 
-    def test_sync_with_included_tags_using_wildcard(self):
-        """Test whether the repository is synced only with included tags that use wildcards."""
-        include_tags = [
-            "ml_iv",
-            "ml_ii",
-            "manifest_a",
-            "manifest_b",
-            "manifest_c",
-            "manifest_d",
-            "manifest_e",
-        ]
-        self.sync_repository_with_filtered_tags(["ml_??", "manifest*"])
-
-        self.assert_synced_tags(include_tags)
-
-    def test_sync_with_complex_filtering(self):
-        """Test sync repository with included and excluded that use wildcards."""
-        include_tags = [
-            "manifest_a",
-            "manifest_c",
-            "manifest_e",
-        ]
-        self.sync_repository_with_filtered_tags(
-            include_tags=["manifest_*"], exclude_tags=["*_[bd]"]
-        )
+    assert repository.latest_version_href == latest_version_href
 
-        self.assert_synced_tags(include_tags)
+    # Ensure that sync does not create orphan tags
+    assert {tag.pulp_href for tag in first_sync_tags} == {tag.pulp_href for tag in second_sync_tags}
```

Comment:
If pulp didn't create a new version, this check seems redundant.

Reply:
But what about `orphan tags`?

Reply:
Actually, @mdellweg is right here. The orphaned tags will not be visible since we filter tags by a referencing repository. It was my suggestion to add the check. Looking back at this, I regret it.

At this point, we could also consider removing all the residues from `test_sync_idempotency`, along with this check and `container_tag_api.list` calls. It does not make sense to test orphans when we run all the tests in parallel.

Reply:
and residual (orphaned) content is always ok to have in the parallel tests. That is why we optionally clean out orphans before (non-parallel) tests if they really need to start clean.

Reply:
By residues, I meant the old pieces of code that were moved from `test_sync_edimpotency` to `test_sync`.